paddle/optimizer/adadelta_optimizer.cc

```diff
@@ -0,0 +1,36 @@
+#include "adadelta_optimizer.h"
+#include <algorithm>
+#include <cmath>
+
+namespace paddle {
+namespace optimizer {
+
+void AdadeltaOptimizer::set_weight(Tensor* p) {
+  parameter_ = p;
+  size_t size = p->size();
+  accum_gradient_ = new Tensor(size);
```

Comment:
"we never reuse the instance" does not prevent client user to reuse the instance. The only way to prevent user to reuse the instance is to through API: remove `set_weight`, and put weight initialization into constructor. User can not reuse the instance only when there is no API for doing so.